Markdown/Markdown.sublime-syntax

```diff
@@ -418,9 +418,10 @@ contexts:
       push:
         - meta_scope: markup.heading.1.markdown
         - meta_content_scope: entity.name.section.markdown
-        - match: \s*(#*)$\n?
+        - match: \s*(#*)$(\n)
```

Comment:
hmm, looks like this is failing [example 43 (trailing whitespace)](http://spec.commonmark.org/0.27/#example-43) atm, any chance you could fix for it and a test assertion while you are here please? :)

Reply:
Looking into it!

Reply:
Regarding the `\n` not being optional, I'm not sure why it wouldn't work, but have you tried having the `?` outside the capture group like `(\n)?`? I guess it is not that important if it is no longer optional as a header is extremely unlikely to be on the last line of a document...

I'm half thinking of suggesting moving the whole

```yaml
       - match: \s*(#*)$(\n)?
          captures:
            1: punctuation.definition.heading.end.markdown
            2: meta.whitespace.newline.markdown
          pop: true
        - include: inline-bold-italic
```

block to a separate context (and including it from the current locations), to avoid repetition, but I'm not sure if it's worth it...

Reply:
> have you tried having the ? outside the capture group like (\n)??

jup

> I'm half thinking of suggesting moving the whole

probably not a bad idea... it's already not-so-easy-as-I-thought, so I don't mind putting in some more effort. 

Am I understanding correctly that before this PR it's also failing example 43?

Reply:
>  a header is extremely unlikely to be on the last line of a document...

Well, it kinda matters if you're writing a new document and the first line is a h1. And that's actually very likely.

There are less elegant ways to make the \n optional, so I think I got something. Commit coming up.

Reply:
Yes it was/is failing example 43 before this PR, sorry for not being clearer

Reply:
@keith-hall I'm having more luck with `[ ]` as apposed to `\s`, with regards to making the `\n` optional. The spec also allows only spaces, not tabs or any other whitespace. Not sure how you feel about such a change?